In [1]:
import csv
import datetime

f = open("guns.csv", "r")
data = list(csv.reader(f))

print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
headers = data[0]

data = data[1:]

print(data[0:5])
print(headers)

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]
['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']


In [3]:
years = []
for each in data:
    years.append(each[1])
    

In [4]:
print(years[0:10])


['2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012', '2012']


In [5]:
year_counts = {}
for each in years:
    if each in year_counts:
        year_counts[each] = year_counts[each] + 1
    else:
        year_counts[each] = 1

print(year_counts)

{'2014': 33599, '2013': 33636, '2012': 33563}


In [6]:
dates = []
dates = [datetime.datetime(year = int(row[1]), month = int(row[2]), day = 1)for row in data[0:]]
#for each in data:
    #dates.append(int(each[1]))
    #dates.append(int(each[2]))
    #dates.append(datetime(year = int(each[1]), month = int(each[2]), day = 1)
print(dates[0:20])    

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 3, 1, 0, 0), datetime.datetime(2012, 4, 1, 0, 0)]


In [7]:
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1
print(date_counts)

{datetime.datetime(2012, 4, 1, 0, 0): 2795, datetime.datetime(2012, 8, 1, 0, 0): 2954, datetime.datetime(2012, 6, 1, 0, 0): 2826, datetime.datetime(2014, 6, 1, 0, 0): 2931, datetime.datetime(2013, 2, 1, 0, 0): 2375, datetime.datetime(2014, 3, 1, 0, 0): 2684, datetime.datetime(2013, 6, 1, 0, 0): 2920, datetime.datetime(2013, 1, 1, 0, 0): 2864, datetime.datetime(2012, 1, 1, 0, 0): 2758, datetime.datetime(2012, 2, 1, 0, 0): 2357, datetime.datetime(2013, 12, 1, 0, 0): 2765, datetime.datetime(2013, 3, 1, 0, 0): 2862, datetime.datetime(2014, 12, 1, 0, 0): 2857, datetime.datetime(2014, 2, 1, 0, 0): 2361, datetime.datetime(2012, 5, 1, 0, 0): 2999, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2013, 9, 1, 0, 0): 2742, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2014, 8, 1, 0, 0): 2970, datetime.datetime(2013, 11, 1, 0, 0): 2758, datetime.datetime(2013, 8, 1, 0, 0): 2859, datetime.datetime(2014, 5, 1, 0, 0): 2864, datetime.datetime(2014, 9, 1, 0, 0): 2914, dateti

In [8]:
sex_counts = {}

for row in data:
    if row[5] in sex_counts:
        sex_counts[row[5]] = sex_counts[row[5]] + 1
    else:
        sex_counts[row[5]] = 1
print(sex_counts)


{'M': 86349, 'F': 14449}


In [9]:
race_counts = {}

for row in data:
    if row[7] in race_counts:
        race_counts[row[7]] = race_counts[row[7]] + 1
    else:
        race_counts[row[7]] = 1
print(race_counts)

{'Native American/Native Alaskan': 917, 'Black': 23296, 'Asian/Pacific Islander': 1326, 'White': 66237, 'Hispanic': 9022}


So far in examining this data on US gun deaths by gender and race it seems so far that males are far more involed with gun deaths.  And further by race - white are the majority of deaths with a total of 66,237.  Second behind white folks, come blacks with a total of 23,296 deaths.  Also appears to be slightly higher rates of deaths in summer months but will definitely need more examination.  


In [10]:
f = open("census.csv", "r")
census = list(csv.reader(f))

print(census)


[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [11]:
mapping = {
    "Asian/Pacific Islander" : 15834141 ,
    "Black" : 40250635,
    "Native American/Native Alaskan" : 3739506 ,
    "Hispanic" : 44618105,
    "White" : 197318956
}

print(mapping)
race_per_hundredk = {}

for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000
    
print(race_per_hundredk)
#for row in race_counts:
    #race_per_hundredk[row] = race_per_hundredk[race_counts[row]/mapping[row] * 100000]
    
#print(race_per_hundredk)






{'Native American/Native Alaskan': 3739506, 'Black': 40250635, 'Asian/Pacific Islander': 15834141, 'White': 197318956, 'Hispanic': 44618105}
{'Native American/Native Alaskan': 24.521955573811088, 'Black': 57.8773477735196, 'Asian/Pacific Islander': 8.374309664161762, 'White': 33.56849303419181, 'Hispanic': 20.220491210910907}


In [12]:
intents = [each[3] for each in data]
print(intents[0:10])

['Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide', 'Undetermined', 'Suicide', 'Accidental', 'Suicide']


In [13]:
races = [each[7] for each in data]
print(races[0:10])

['Asian/Pacific Islander', 'White', 'White', 'White', 'White', 'Native American/Native Alaskan', 'White', 'Native American/Native Alaskan', 'White', 'Black']


In [14]:
homicide_race_counts = {}


for i,race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1
#print(homicide_race_counts)

race_per_hundredk = {}
for k,v in homicide_race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000

print(race_per_hundredk)
    

{'Native American/Native Alaskan': 8.717729026240365, 'Black': 48.471284987180944, 'Asian/Pacific Islander': 3.530346230970155, 'White': 4.6356417981453335, 'Hispanic': 12.627161104219914}


In [ ]:
It does appear that Black individuals have a higher incidence rate in homicides as well as in deaths by gun in the US based on the census data and comparisons by using ratios per 100,000 people.  For further analysis you can examine by months or by rates by gender and race.  